src/models/automl/shap_select_feature_selector.py

```
@@ -0,0 +1,224 @@
+from shap_select import shap_select
+from lightgbm import LGBMRegressor
+
+from src.models.automl.autogluon_trainer import top_dataframes
+from src.utils.cleaned_data_cache import collect_garbage
+from src.utils.logger import get_logger
+from src.models.common.training_data_prep import prepare_common_training_data
+
+logger = get_logger(__name__)
+
+# Local metric replicated from project to remain compatible with AutoGluon
+def conditional_mean_absolute_error(y_true, y_pred, *args, **kwargs):
```

Ruff (0.14.0)

Unused function argument: `args`

(ARG001)